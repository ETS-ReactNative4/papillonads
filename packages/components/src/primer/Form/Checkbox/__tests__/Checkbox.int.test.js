import { renderCustom, renderIsChecked, renderIsNotChecked } from './Checkbox.int.render'

describe('<Checkbox />', () => {
  const onChangeMockFn = jest.fn()

  afterEach(() => jest.clearAllMocks())

  describe('Event', () => {
    test('must handle event when onChange()', () => {
      const mountCustomRender = global.renderMount(renderCustom(onChangeMockFn))
      mountCustomRender.find('input').simulate('change')
      expect(mountCustomRender.props().onChange).toBe(onChangeMockFn)
      expect(mountCustomRender.props().onChange).toHaveBeenCalledWith(expect.any(Object))
    })
  })

  describe('Render', () => {
    test('must match IsChecked()', () => {
      expect(global.renderToJSON(renderIsChecked())).toMatchSnapshot()
    })

    test('must match renderIsNotChecked()', () => {
      expect(global.renderToJSON(renderIsNotChecked())).toMatchSnapshot()
    })
  })
})
