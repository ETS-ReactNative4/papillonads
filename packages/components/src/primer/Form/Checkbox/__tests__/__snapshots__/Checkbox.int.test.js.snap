// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Render must match IsChecked() 1`] = `
<div
  className="m-0 form-checkbox"
>
  <label
    htmlFor="checkbox"
  >
    <input
      aria-describedby="help-text-for-checkbox"
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    is checked
  </label>
</div>
`;

exports[`<Checkbox /> Render must match renderIsNotChecked() 1`] = `
<div
  className="m-0 form-checkbox"
>
  <label
    htmlFor="checkbox"
  >
    <input
      aria-describedby="help-text-for-checkbox"
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    is not checked
  </label>
</div>
`;
